<?xml version="1.0"?>
<launch>
  <!-- This file launches all the vision and recognition nodes. Assumes that the cameras are running. -->

  <!-- Mode of object detector -->
  <!-- Continuous streaming is expensive and meant for debugging. -->
  <arg name="pth_file_name"		default="WRS_day3.pth"/>
  <arg name="continuous_streaming"	default="false"/>
  <arg name="camera_name"		default="a_bot_outside_camera"/>
  <arg name="vis"			default="true"/>
  <arg name="use_nodelet"		default="false"/>
  <arg if="$(arg use_nodelet)"
       name="manager"			value="camera_manager"/>
  <arg unless="$(arg use_nodelet)"
       name="manager"			value=""/>

  <!-- Localization parameters -->
  <arg name="nposes"	default="2"/>
  <arg name="timeout"	default="10"/>
  <arg name="ply_dir"	default="$(find o2ac_parts_description)/meshes"/>
  <arg name="urdf_dir"	default="$(find o2ac_parts_description)/urdf/generated"/>

  <rosparam param="shaft_hole_detection/bbox">[375, 278, 90, 90]</rosparam>

  <!-- Construct recognition pipeline -->
  <include file="$(dirname)/recognition_pipeline.launch">
    <arg name="manager"			value="$(arg manager)"/>
    <arg name="pth_file_name"			value="$(arg pth_file_name)"/>
    <arg name="continuous_streaming"	value="$(arg continuous_streaming)"/>
    <arg name="nposes"			value="$(arg nposes)"/>
    <arg name="timeout"			value="$(arg timeout)"/>
    <!-- <arg name="ply_dir"			value="$(arg ply_dir)"/> -->
    <arg name="urdf_dir"		value="$(arg urdf_dir)"/>

    <!-- Topic names for a raw realsense camera -->
    <arg name="camera_info_topic" value="$(arg camera_name)/color/camera_info"/>
    <arg name="image_topic"	  value="$(arg camera_name)/color/image_raw"/>
    <arg name="depth_topic"	  value="$(arg camera_name)/aligned_depth_to_color/image_raw"/>

    <!-- Topic names of the camera multiplexer -->
    <!-- <arg name="camera_info_topic" value="$(arg camera_name)/camera_info"/>
    <arg name="image_topic"	  value="$(arg camera_name)/image"/>
    <arg name="depth_topic"	  value="$(arg camera_name)/depth"/> -->
    <arg name="normal_topic"	  value=""/>
  </include>

  <group if="$(arg vis)">
    <!-- Separate node for Python3 functions -->
    <node name="o2ac_py3_vision_server"
    pkg="o2ac_vision" type="o2ac_py3_vision_server.py"/>

    <!-- Spawn visualizer and parameter configurator -->
    <node name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
          args="-d $(find o2ac_vision)/launch/pipeline_test.rviz"/>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
  </group>

</launch>
